src/test/java/com/flipkart/zjsonpatch/AbstractTest.java

```diff
@@ -1,88 +1,54 @@
 
-    protected AbstractTest(String fileName) throws IOException {
-        String path = "/testdata/" + fileName + ".json";
-        InputStream resourceAsStream = JsonDiffTest.class.getResourceAsStream(path);
-        String testData = IOUtils.toString(resourceAsStream, "UTF-8");
-        JsonNode testsNode = objectMapper.readTree(testData);
-        jsonNode = (ArrayNode) testsNode.get("ops");
-        errorNode = (ArrayNode) testsNode.get("errors");
-    }
+    @Parameter
+    public PatchTestCase p;
 
     @Test
-    public void testPatchAppliedCleanly() throws Exception {
-        for (int i = 0; i < jsonNode.size(); i++) {
-            JsonNode first = jsonNode.get(i).get("node");
-            JsonNode second = jsonNode.get(i).get("expected");
-            JsonNode patch = jsonNode.get(i).get("op");
-            String message = jsonNode.get(i).has("message") ? jsonNode.get(i).get("message").toString() : "";
-
-            System.out.println("Test # " + i);
-            System.out.println(first);
-            System.out.println(second);
-            System.out.println(patch);
-
-            JsonNode secondPrime = JsonPatch.apply(patch, first);
-            System.out.println(secondPrime);
-            Assert.assertThat(message, secondPrime, equalTo(second));
+    public void test() throws Exception {
+        if (p.isOperation()) {
+            testOpertaion();
+        } else {
+            testError();
         }
     }
 
-//    @Test
-//    public void testPatchSyntax() throws Exception {
-//        for (int i = 0; i < jsonNode.size(); i++) {
-//            JsonNode first = jsonNode.get(i).get("node");
-//            JsonNode second = jsonNode.get(i).get("expected");
-//            JsonNode patch = jsonNode.get(i).get("op");
-//
-//            System.out.println("Test # " + i);
-//            System.out.println(first);
-//            System.out.println(second);
-//            System.out.println(patch);
-//
-//            JsonNode diff = JsonDiff.asJson(first, second);
-//            System.out.println(diff);
-//            Assert.assertThat(diff, equalTo(patch));
-//        }
-//    }
+    private void testOpertaion() throws Exception {
+        JsonNode node = p.getNode();
 
-    @Test(expected = JsonPatchApplicationException.class)
-    public void testErrorsAreCorrectlyReported() {
-        for (int i = 0; i < errorNode.size(); i++) {
-            JsonNode first = errorNode.get(i).get("node");
-            JsonNode patch = errorNode.get(i).get("op");
+        JsonNode first = node.get("node");
+        JsonNode second = node.get("expected");
+        JsonNode patch = node.get("op");
+        String message = node.has("message") ? node.get("message").toString() : "";
 
-            System.out.println("Error Test # " + i);
-            System.out.println(first);
-            System.out.println(patch);
+        JsonNode secondPrime = JsonPatch.apply(patch, first);
 
-            JsonNode secondPrime = JsonPatch.apply(patch, first);
```

Comment:
Here is the problem with error test cases: once first test case fails with expected exception loop iteration stops and further tests cases are not evaluated
 Add a line note